### PR TITLE
Add config key to prevent glide from prepending the site root

### LINKED ISSUE #1304: Multi-site locale in asset path => file not found
My sites config:
```
    'sites' => 
    [
        'de' => [
            'name' => config('app.name') . ' - DE',
            'locale' => 'de_DE',
            'url' => '/de/',
        ],
        'en' => [
            'name' => config('app.name') . ' - EN',
            'locale' => 'en_US',
            'url' => '/en/',
        ]
    ]
```
The asset path contains now the locale and the file could not be found.
```
https://domain.tld/de/path/to/file.jpg
```

### PR DESCRIPTION
**Pull request by AndreasSchantl**
This PR closes #1304

### CODE REVIEW

File: config/system.php [hunk 1]
```diff
@@ -91,5 +91,6 @@
     'php_max_execution_time' => '-1',
     'ajax_timeout' => '600000',
     'pcre_backtrack_limit' => '-1',
+    'glide_prepend_siteroot' => true,
```
Comment (jasonvarga) [inline config/system.php:94]: Please move this into the `config/assets.php` file.
I think a good place would to name it `prepend_site_root` inside the `image_manipulation` array. Right after `route`. Here: https://github.com/statamic/cms/blob/master/config/assets.php#L18

File: src/Imaging/GlideUrlBuilder.php [hunk 2]
```diff
@@ -37,13 +37,13 @@ public function build($item, $params, $filename = null)
...
-                $path = 'http/'.base64_encode($item);
+                $path = 'http/' . base64_encode($item);
                 break;
             case 'asset':
-                $path = 'asset/'.base64_encode($this->item->containerId().'/'.$this->item->path());
+                $path = 'asset/' . base64_encode($this->item->containerId() . '/' . $this->item->path());
                 break;
             case 'id':
-                $path = 'asset/'.base64_encode(str_replace('::', '/', $this->item));
+                $path = 'asset/' . base64_encode(str_replace('::', '/', $this->item));
```
Comment (jasonvarga) [inline src/Imaging/GlideUrlBuilder.php:46]: Please remove these changes.

File: src/Imaging/GlideUrlBuilder.php [hunk 3]
```diff
@@ -58,6 +58,10 @@ public function build($item, $params, $filename = null)
             $path .= Str::ensureLeft(URL::encode($filename), '/');
         }
 
-        return URL::prependSiteRoot($builder->getUrl($path, $params));
+        if (config('statamic.site.glide_prepend_siteroot')) {
```
Comment (jasonvarga) [inline src/Imaging/GlideUrlBuilder.php:61]: This is pointing to the wrong config value.

After changing the config location in my other comment, this should be `statamic.assets.image_manipulation.prepend_site_root`

**Comment by AndreasSchantl** [comment]
Alright, I did change everything to match your suggestions.

**Comment by AndreasSchantl** [comment]
Did you try clearing the cache after that? @Pfeifakopf

**Comment by Pfeifakopf** [comment]
@AndreasSchantl Yes, i cleared the cache with optimize:clear and the image cache separately again. The paths are still wrong.

**Comment by AndreasSchantl** [comment]
@jasonvarga might this be merged with 3.0.2?

**Comment by jonassiewertsen** [comment]
Am uprading a page from version 2 to 3 right now and running into this again. This PR merged would be really nice :-)

**Comment by afonic** [comment]
I agree, this breaks assets for a multi-site setup if I'm not mistaken.

**Comment by FlorianMoser** [comment]
I'm also stuck with this. Of course, you can create a symlink from `/public/{locale}/{cache_path}` to `/public/{cache_path}` as suggested [in the docs](https://statamic.dev/tags/glide#using-glide-with-locales). But then you cannot access the URL `/{locale}` anymore, because this is now an existing (but empty) directory.

**Comment by AndreasSchantl** [comment]
Closed by #2379

**Pull request closed without merging**